frontends/infinite-corridor/src/test-utils/mockAxios.ts

```diff
@@ -48,21 +50,22 @@ const mockRequest = (
   responseBody: unknown = undefined,
   code: number
 ) => {
-  if (code >= 400) {
-    when(makeRequest)
-      .calledWith(method, url, requestBody)
-      .mockRejectedValue({
-        response: {
-          data:   responseBody,
-          status: code
-        }
-      })
-  } else {
-    when(makeRequest).calledWith(method, url, requestBody).mockResolvedValue({
-      data:   responseBody,
-      status: code
+  when(makeRequest)
+    .calledWith(method, url, requestBody)
+    .mockImplementation(async () => {
+      const data = await responseBody
```

Comment:
Previously, when `responseBody` was a promise, our mock API was responding immediately with a response like `{ status: 200, data: Promise<pending> }`. This was incorrect.

Now the respond is `Promise<pending>`, and resolves to `{ status: 200, data }`.